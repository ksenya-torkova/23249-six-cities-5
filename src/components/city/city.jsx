import {cityTypes} from "../../prop-types";
import React from "react";

const City = (props) => {
  const {city, isCurrent, onCityClick} = props;

  return (
    <li className="locations__item">
      <a
        className={`locations__item-link tabs__item ${isCurrent ? `tabs__item--active` : ``}`}
        href="#"
        onClick = {onCityClick}
      >
        <span>{city}</span>
      </a>
    </li>
  );
};

City.propTypes = cityTypes;

export {City};
export default React.memo(City);
