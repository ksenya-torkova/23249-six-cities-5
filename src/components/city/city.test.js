import React from "react";
import renderer from "react-test-renderer";
import {City} from "./city";

const noop = () => {};

it(`Should City render correctly`, () => {
  const tree = renderer
    .create(
        <City
          city={`Amsterdam`}
          isCurrent={false}
          onCityClick={noop}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
