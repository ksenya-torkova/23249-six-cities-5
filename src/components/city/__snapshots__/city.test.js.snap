// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should City render correctly 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item "
    href="#"
    onClick={[Function]}
  >
    <span>
      Amsterdam
    </span>
  </a>
</li>
`;
